src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBuilder.java

```diff
@@ -247,6 +308,38 @@ public void searchWorkOrder()
         }
     }
 
+    /**
+     * Sets the work order with the given id as the work order for this buildings citizen.
+     * 
+     * @param orderId the id of the work order to select.
+     */
+    public void setWorkOrder(int orderId)
+    {
+        final ICitizenData citizen = getMainCitizen();
+        if (citizen == null)
+        {
+            return;
+        }
+
+        if (citizen.getJob(JobBuilder.class).hasWorkOrder()) {
```

Comment:
bracket on next line

Reply:
should probably cancel the active workorder here, if its not the same as the new one. As I think the player would expect to be able to directly switch to the workorder right away, not waiting for current to finish

Reply:
gotta make sure to reset all the data then too regarding the workorder (else the AI will get messy)

Reply:
ye atm it only listens to workerorder.cancel, maybe we cancel the workorder and make a new one with similar content?

Reply:
Currently the gui for selecting a work order doesn't show while one is already selected, this is just to prevent spam clicking from causing issues.
I also did never run into issues without this, but i wanted to make sure.

Reply:
sounds good for me then

Reply:
so if you toggle manual while he is building sth just nothing happens?
I think the main part where ppl do use this is when "Oh Builder X took task Y, but I wanted him to do Z"

Reply:
Are you able to add text to state "just finishing this job gov'ner" so players don't get concerned when they try to change task

Reply:
Yeah, on switching to manual we could print a message with "Succesfully switched, iwll just finish the current task"

Reply:
Is just a chat message ok, or should the citizen somehow say that in his gui?
Also should he tell that to all players in the colony, or just the player switching it?

Reply:
chat message is fine, just for the switching player

Reply:
OK, i will add that tomorrow

Reply:
I added a message saying `BUILDER NAME: Succesfully switched to manual mode, i will just finish the current task.` for this.
The text of the message isn't great, but i don't know how to make it better without making it significantly longer.

Reply:
@ravenbuilder934